src/adr/0034-mailbox-api-visibility-and-usage.md

```diff
@@ -0,0 +1,39 @@
+# 34. Mailbox API visibility and usage
+
+Date: 2020-04-27
+
+## Status
+
+Accepted (lazy consensus)
+
+## Context
+
+For some reasons, all mailboxes implementations rely on some abstract classes located in `mailbox-store` module (like 
+`StoreMailboxManager` and the `Mapper`s classes). For example, a `CassandraMailboxmanager` has to extend `StoreMailboxManager` 
+(that implements `Mailboxmanager` from the `mailbox-api`) that requires the implementation of some `Mapper`s.
+
+`Mapper`s are designed to provide low-level functions and methods on mailboxes. It's recurrent that we are tempted in 
+James, outside of the `mailbox` modules, to rely on some of those classes located in `mailbox-store` to have an easier 
+access on some user's mailboxes or messages. 
+
```

Comment:
Is there some services handled at the manager layer that are not covered by the mapper layer?

Reply:
The manager layer do further validation including right checking, event dispatching (resulting in message search index indexing, current quota calculation mainly), quota validation. Not relying on the manager layer is thus error prone and can lead to security vulnerabilities.